### PR TITLE
data and ids implementations

### PR DESCRIPTION
**Pull request by muku314115**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and please add links to any
relevant follow-up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json)
  in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (
  see [PR Targeting](https://github.com/AssetMantle/modules/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines
  for [building modules](https://github.com/AssetMantle/modules/blob/master/docs/building-modules)
- [ ] included the necessary unit and
  integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add your handle next to the items
reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the
  correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by avkr003 — APPROVED** [review]

File: schema/data/base/idData.go [hunk 1]
```diff
@@ -8,63 +8,57 @@ import (
...
 }
-func (idData idData) String() string {
-	return idData.Value.String()
+func (idData *Data_IdData) String() string {
+	return idData.IdData.String()
 }
-func (idData idData) Bytes() []byte {
-	return idData.Value.Bytes()
+func (idData *Data_IdData) Bytes() []byte {
+	return idData.Bytes()
```
Comment (deepanshutr) [inline schema/data/base/idData.go:37]: idData.idData.Bytes()

File: schema/ids/base/assetID.go [hunk 2]
```diff
@@ -8,52 +8,50 @@ import (
...
 }
-func assetIDFromInterface(i interface{}) assetID {
-	switch value := i.(type) {
-	case assetID:
-		return value
-	default:
+func (assetID *ID_AssetID) IsOwnableID() {}
+func (assetID *ID_AssetID) IsAssetID()   {}
+func (assetID *ID_AssetID) Bytes() []byte {
+	return assetID.AssetID.HashID.IdBytes
```
Comment (deepanshutr) [inline schema/ids/base/assetID.go:26]: assetID.AssetID.Bytes

File: schema/ids/base/dataID.go [hunk 3]
```diff
@@ -5,80 +5,65 @@ package base
...
-	switch value := i.(type) {
-	case dataID:
-		return value
-	default:
+
+func GenerateDataID(data data.Data) ids.ID {
+	if data == nil {
 		panic(errorConstants.MetaDataError)
 	}
+	return NewDataID(data.GetType(), data.GenerateHashID())
```
Comment (deepanshutr) [inline schema/ids/base/dataID.go:36]: should be data,GetTypeID

File: schema/ids/base/dataID.go [hunk 4]
```diff
@@ -5,80 +5,65 @@ package base
...
 }
-
-func NewDataID(data data.Data) ids.DataID {
-	if data == nil {
+func NewDataID(Type ids.ID, hashID ids.ID) ids.ID {
```
Comment (deepanshutr) [inline schema/ids/base/dataID.go:38]: typeID

**Review by deepanshutr — APPROVED** [review]
reviewed.

**Pull request merged**